tests/test_auth.py

```diff
@@ -0,0 +1,47 @@
+""" Test case for sign in and sign up """
+import pytest
+import allure
+from data_tests.auth import REGISGER_DATA, TOKEN
+from utilities.db import change_varification_link
+from base import LOGIN_URL, LOGOUT_URL, USER_CONFIRM_EMAIL_URL, USER_REGISTER_URL
+
+
+@allure.story('Sign In')
+@pytest.mark.parametrize('user, password, expected', [
+    ('admin@gmail.com', 'admin', 'ROLE_ADMIN'),
+    ('user@gmail.com', 'user', 'ROLE_USER'),
+    ('cowner@gmail.com', 'cowner', 'ROLE_COWNER')
+])
+def test_login(app, user, password, expected):
+    """ Test for authenticate and authorize user.
+    doesn't use application method, to check auth step by step"""
+    session = app.session
+    response = session.post(LOGIN_URL, auth=(user, password))
+    with allure.step("Check if status code equal 200"):
+        assert response.status_code == 200, "Wrong status code"
+    data = response.json()
+    with allure.step("Check if username is correct"):
+        assert data['username'] == user, "Wrong username"
+    with allure.step("Check if user authorized correctly"):
+        assert data['authorities'][0]['authority'] == expected, "Wrong user role"
+    session.get(LOGOUT_URL)
+
+
+@allure.feature('Sign Up')
+def test_sign_up_begin(app):
+    """ Start to register user. Up to send email """
+    app.post(USER_REGISTER_URL, data=REGISGER_DATA)
+    app.check_200()
```

Comment:
why is there used a method for checking status_code, but before this assertion was in the test?

Reply:
fixed